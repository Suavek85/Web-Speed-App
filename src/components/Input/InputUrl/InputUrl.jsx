import React from 'react'
import './InputUrl.scss'

function InputCard() {
  return (
    <label htmlFor="urlinput" className="urlinput">
      <input autoComplete='off' type="url" id="urlinput" placeholder="&nbsp;" />
      <span className="label">Enter your website url</span>
      <span className="border"></span>
    </label>
  )
}

export default InputCard