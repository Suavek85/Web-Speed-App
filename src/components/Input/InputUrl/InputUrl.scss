.urlinput {
  display: flex;
  position: relative;
  width: 100%;
  max-width: 280px;
  margin-top: 30px;
  margin-bottom: 30px;
}
.urlinput .label {
  position: absolute;
  top: -5px;
  left: 0;
  font-size: 14px;
  color:  #F76C6C;
  font-weight: 500;
  transform-origin: 0 0;
  transition: all 0.2s ease;
}
.urlinput .border {
  position: absolute;
  bottom: 0;
  left: 0;
  height: 2px;
  width: 100%;
  background: #F76C6C;
  transform: scaleX(0);
  transform-origin: 0 0;
  transition: all 0.15s ease;
}
.urlinput input {
  -webkit-appearance: none;
  width: 100%;
  border: 0;
  font-family: inherit;
  padding: 8px 0;
  height: 36px;
  font-size: 16px;
  font-weight: 500;
  border-bottom: 2px solid rgb(168, 193, 230);
  background: none;
  border-radius: 0;
  color: #223254;
  transition: all 0.15s ease;
}

.urlinput input:not(:placeholder-shown) + span {
  color: #F76C6C;
  transform: translateY(-26px) scale(0.75);
}
.urlinput input:focus {
  background: none;
  outline: none;
}
.urlinput input:focus + span {
  transform: translateY(-26px) scale(0.75);
}
.urlinput input:focus + span + .border {
  transform: scaleX(1);
}
