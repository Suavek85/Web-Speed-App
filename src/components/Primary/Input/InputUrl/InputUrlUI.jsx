import React from 'react'

import { INPUT_HINT } from '../../../../constants/content'
import './InputUrlUI.scss'
import styles from './InputUrlUI.scss'

export default function InputUrl() {

  return (
    <label 
      htmlFor="urlinput" 
      className={ styles.block }>
      <input 
        autoComplete='off' 
        type="url" 
        id="urlinput" 
        autoFocus 
        placeholder="&nbsp;" 
      />
      <span className={ styles.blockLabel } >
        { INPUT_HINT }
      </span>
      <span className={ styles.blockBorder } ></span>
    </label>
  )
}
