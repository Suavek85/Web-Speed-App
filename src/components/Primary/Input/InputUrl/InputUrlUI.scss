@import '../../../../scss/styles';

$block: url-input;
$blockLabel: elem($block, label);
$blockBorder: elem($block, border);

.#{$block} {
  position: relative;
  display: flex;
  width: 100%;
  max-width: 210px;

  &__label {
    position: absolute;
    top: -5px;
    left: 0;
    font-size: em(14);
    font-weight: 500;
    color: $color-alert;
    transition: all 0.2s ease;
    transform-origin: 0 0;
  }

  &__border {
    position: absolute;
    bottom: 0;
    left: 0;
    width: 100%;
    height: 2px;
    background: $color-alert;
    transition: all 0.15s ease;
    transform: scaleX(0);
    transform-origin: 0 0;
  }
}

input {
  width: 100%;
  height: 36px;
  padding: 8px 0;
  font-family: inherit;
  font-size: em(16);
  font-weight: 500;
  color: $color-small-font;
  background: none;
  border: 0;
  border-bottom: 2px solid $color-light-blue;
  border-radius: 0;
  transition: all 0.15s ease;
  -webkit-appearance: none;
}

input:not(:placeholder-shown) + span {
  color: $color-alert;
  transform: translateY(-26px) scale(0.75);
}

input:focus {
  background: none;
  outline: none;
}

input:focus + span {
  transform: translateY(-26px) scale(0.75);
}

input:focus + span + .border {
  transform: scaleX(1);
}

:export {
  block: $block;
  blockLabel: $blockLabel;
  blockBorder: $blockBorder;
}
