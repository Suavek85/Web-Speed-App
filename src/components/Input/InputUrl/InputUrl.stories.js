import React from 'react'
import { storiesOf } from '@storybook/react'
import InputUrl from './InputUrl'

const stories = storiesOf('UI Components', module)

stories.add('Input Url', () => {
  return <InputUrl />
})