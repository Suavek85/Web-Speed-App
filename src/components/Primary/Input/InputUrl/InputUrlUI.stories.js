import React from 'react'
import { storiesOf } from '@storybook/react'
import InputUrlUI from './InputUrlUI'

const stories = storiesOf('UI Components', module)

stories.add('Input Url', () => {
  return <InputUrlUI />
})