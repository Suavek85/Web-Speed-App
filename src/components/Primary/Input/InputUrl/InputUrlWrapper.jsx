import React from 'react'
import InputUrlUI from './InputUrlUI'

export default function InputUrl() {
  return (
    <InputUrlUI  />
  )
}
