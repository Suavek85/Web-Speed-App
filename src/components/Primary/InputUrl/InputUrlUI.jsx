import React, { useRef } from 'react'

import { INPUT_HINT } from '../../../constants/content'
import './InputUrlUI.scss'
import styles from './InputUrlUI.scss'

export default function InputUrl() {

  const myTextInput = useRef('')

  return (
    <label 
      htmlFor="urlinput" 
      className={ styles.block }>
      <input 
        autoComplete='off' 
        type="url" 
        id="urlinput" 
        autoFocus 
        placeholder="&nbsp;" 
        ref={myTextInput}
      />
      <span className={ styles.blockLabel } >
        { INPUT_HINT }
      </span>
      <span className={ styles.blockBorder } ></span>
    </label>
  )
}
