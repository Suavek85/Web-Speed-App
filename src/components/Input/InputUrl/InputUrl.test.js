import React from 'react'
import { shallow } from 'enzyme'
import InputUrl from './InputUrl'

it('Should match the snapshot', () => {
  const wrapper = shallow(<InputUrl />)
  expect(wrapper).toMatchSnapshot()
})