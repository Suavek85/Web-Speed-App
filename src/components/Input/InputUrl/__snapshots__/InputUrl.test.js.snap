// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match the snapshot 1`] = `
<label
  className="urlinput"
  htmlFor="urlinput"
>
  <input
    autoComplete="off"
    id="urlinput"
    placeholder=" "
    type="url"
  />
  <span
    className="label"
  >
    Enter your website url
  </span>
  <span
    className="border"
  />
</label>
`;
